src/java.base/share/classes/java/util/concurrent/ExecutorService.java

```diff
@@ -135,7 +135,7 @@
  * @since 1.5
  * @author Doug Lea
  */
-public interface ExecutorService extends Executor {
+public interface ExecutorService extends Executor, AutoCloseable {
```

Comment:
The class documentation should be expanded to explain that an ExecutorService can be used with a try-with-resources

Reply:
Most AutoCloseables do not mention this in class-level javadocs. Is there a reason you think this one should?

Reply:
close() is now equivalent to the method shutdownAndAwaitTermination() shown in the javadoc so i believe , replacing it with a try-with-resources is better in term of documenting how an executor service can be used

Reply:
OK, done (see https://bugs.openjdk.java.net/browse/JDK-8285450)